wordpress-mobile/WordPress-Android
# [Nullability Annotations to Java Classes] Add Missing Nullability Annotations to All SDK Override `onNewIntent(...)` Methods (`risky`)

Parent #18909

This PR adds the missing `@NonNull` annotation to all SDK override `onNewIntent(...)` related method on `Activity` classes, including updating any Kotlin related classes that had that as `@Nullable` via the use of the `Intent?` declaration.

FYI: This change is `risky`, meaning that there are compile-time changes associated with this change and thus needs testing to verify correctness.

-----

Nullability Annotation List:

1. [Add missing nn-a to on new intent methods for activities](https://github.com/wordpress-mobile/WordPress-Android/commit/2adc6dbbfaeb4f354e46566b277545d7f28e2d3f)

Lint Warnings Suppression List:

1. [Suppress non constant res id lint warning for media browser](https://github.com/wordpress-mobile/WordPress-Android/commit/e65f780ff0f755c3bc1ad62438efb141c8824765)

Risky List:

1. [Make intent non null on the on new intent functions for kotlin](https://github.com/wordpress-mobile/WordPress-Android/commit/bd62ee79c2ca4c54bc0d0b1a8e4bfb42a1e9afa7)

-----

## To test:

- Smoke test any `onNewIntent` related functionality on both, the WordPress and Jetpack apps, and see if everything is working as expected.
- In addition to the above smoke test, you can expand the below and follow the inner and more explicitly test steps within, which is mainly related to the risky changes that are associated with this PR:

<details>
    <summary>1. Stats Screen [StatsActivity.kt]</summary>

ℹ️ This test applies to the `Jetpack` app.

- Go to `Stats` screen and verify everything works as expected.
- Then, add any `Stats` related widget, like the `All-time` widget, configure and place it.
- Finally, click on that `Stats` widget you just added and via `debug` mode, check the `onNewIntent(...)` function is called and the `Intent` is not `null`.

</details>

<details>
    <summary>2. Scan Screen [ScanActivity.kt + ScanFragment.kt]</summary>

ℹ️ This test applies to the `Jetpack` app.

- Go to `Scan` screen and verify everything works as expected.
- Then, add a new `Threat` to your site, like the `Vulnerable Plugin: elementor (version 3.7.7)` and run a new scan. This new `Threat` should be found and appear on the main screen.
- Finally, click on that `Threat` you just added then fix it. Via `debug` mode, check the `onNewIntent(...)` function is called and the `Intent` is not `null`.

</details>

<details>
    <summary>3. Pages Screen [PagesActivity.kt]</summary>

ℹ️ This test applies to both, the `Jetpack` and `WordPress` app.
⁉️ This flow might not longer be valid as I couldn't make it work.

- Go to `Pages` screen and verify everything works as expected.
- Then, open a `Page` and edit it. Quickly exist the `Pages` screen before the upload is finished.
- Finally, when the `Page` upload is done and the notification is shown, click on that `Notification` and via `debug` mode, check the `onNewIntent(...)` function is called and the `Intent` is not `null`.

FYI: I got the test instruction above from [this](https://github.com/wordpress-mobile/WordPress-Android/pull/8327) PR.
PS: No matter, this test instruction is just informational-only as the `onNewIntent(...)` signature of this `PagesActivity.kt` class was already using a non-null `intent`, with no changes happening here.

</details>

<details>
    <summary>4. Posts Screen [PostsListActivity.kt]</summary>

ℹ️ This test applies to both, the `Jetpack` and `WordPress` app.
⁉️ This flow might not longer be valid as I couldn't make it work.

- Got to `Posts` screen and verify everything works as expected.
- Then, turn off WiFi, create a new `Post` and publish it. Switch site, open the `Posts` screen on that site and turn the WiFi back on.
- Finally, when the `Post` upload is done and the notification is shown, click on that `Notification` and via `debug` mode, check the `onNewIntent(...)` function is called and the `Intent` is not `null`.

FYI: I got the test instruction above from [this](https://github.com/wordpress-mobile/WordPress-Android/pull/9290) PR.
PS: No matter, this test instruction is just informational-only as the `onNewIntent(...)` signature of this `PostsListActivity.kt` class was already using a non-null `intent`, with no changes happening here.

</details>

-----

## Regression Notes

1. Potential unintended areas of impact

    - Potential new NPE start happening on the `Stats` and/or `Scan` screens, always related to the `onNewIntent(...)` functionality, which had its 'intent' parameter updated from being a nullable one to a non-null.

3. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To test` section above.

4. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

## PR submission checklist:


UI Changes testing checklist:
